Maple2.Server.Game/PacketHandlers/MasteryHandler.cs

```diff
@@ -32,12 +32,52 @@ private enum Command : byte {
     public override void Handle(GameSession session, IByteReader packet) {
         var command = packet.Read<Command>();
         switch (command) {
+            case Command.Reward:
+                HandleReward(session, packet);
+                break;
             case Command.Craft:
                 HandleCraft(session, packet);
                 break;
         }
     }
 
+    private void HandleReward(GameSession session, IByteReader packet) {
+        int rewardBoxDetails = packet.ReadInt();
+        var type = (MasteryType) (rewardBoxDetails / 1000);
+        int level = rewardBoxDetails % 100;
+
+        if (session.Player.Value.Unlock.MasteryRewardsClaimed.TryGetValue(rewardBoxDetails, out bool isClaimed) && isClaimed) {
+            session.Send(MasteryPacket.Error(MasteryError.s_mastery_error_unknown));
+            return;
+        }
+
+        if (!TableMetadata.MasteryRewardTable.Entries.TryGetValue(type, out MasteryRewardTable.Entry? entry) || !entry.Levels.TryGetValue(level, out MasteryRewardTable.Level? levelMetadata)) {
```

Comment:
As mentioned above, once you switch to nested dictionaries, the extension method will allow you to do:
```cs
TableMetadata.MasteryRewardTable.Entries.TryGetValue(type, level, out ...);
```